File: qw/langs/py/func1.py
# -*- coding: UTF-8 -*-

'''
#用于快速提取并去重面板内的引用资源
'''

import re

#打开文件
fo = open("e:/newfile", "r")
uniqDict = {}
line = "aaa"

#提取
while line:
    line = next(fo)
    
    if line:
        pattern = re.compile( r'At/[0-9a-zA-Z_/]*' )
        it = re.finditer( pattern, line )
        
        if it:
            for item in it:
                k = item.group().replace('/', '_')
                #去重、输出
                if k not in uniqDict:
                    uniqDict[k] = 'true'
                    print(item.group())

fo.close()

File: qw/langs/java/func1.java
/**
* 用于快速提取并去重面板内的引用资源
*/

import java.io.BufferedReader;
import java.io.FileInputStream;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.regex.*;
import java.util.HashMap;

public class func1 {
    public static void main(String[] args) throws IOException {
        
        //打开文件
        FileInputStream inputStream = new FileInputStream("e:/newfile");
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(inputStream));
        
        String str;
        String pattern = "At[0-9a-zA-Z_/]*";
        Pattern r = Pattern.compile(pattern);
        Matcher m;
        HashMap<String, String> uniqMap = new HashMap<String, String>();
        String k;
        int matcher_start = 0;
        
        //提取
        while((str = bufferedReader.readLine()) != null){
            m = r.matcher(str);
            matcher_start = 0;
            while(m.find(matcher_start)){

                //System.out.println(m.groupCount());
                str = m.group();
                //去重、输出
                k = str.replaceAll("/", "_");
                //System.out.println(k);

                if(uniqMap.get(k) == null){
                    uniqMap.put(k, "true");
                    System.out.println(str);
                }
                matcher_start = m.end();

            }
        }
        //System.out.println(uniqMap);
        inputStream.close();
        bufferedReader.close();
    }
}